server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -297,8 +324,6 @@ private void validate(
                         throw new IllegalStateException(finalErrorMsg);
                     }
 
-                    Version minIndexCompatibilityVersion = currentState.getNodes().getMaxNodeVersion().minimumIndexCompatibilityVersion();
```

Comment:
this validation is now performed as part of prepareInitialClusterState in GatewayMetaState.